### DISCUSSION TITLE
Can we change the bulk action column?, what i have do to add some class on action column?

### DISCUSSION THREAD

**Discussion by cecep31** [Power-Components/livewire-powergrid#787] [Q&A]
`<?php

namespace App\Http\Livewire\PowerGrid;

use andcarpi\Popper\Facades\Popper;
use App\Models\PurchaseRequest;
use Illuminate\Support\Carbon;
use Illuminate\Database\Eloquent\Builder;
use PowerComponents\LivewirePowerGrid\Rules\{Rule, RuleActions};
use PowerComponents\LivewirePowerGrid\Traits\ActionButton;
use PowerComponents\LivewirePowerGrid\{Button, Column, Exportable, Footer, Header, PowerGrid, PowerGridComponent, PowerGridEloquent};

final class PurchaseRequestTable extends PowerGridComponent
{
    use ActionButton;

    public $pr_no, $slug, $pr_type, $project_id, $warehouse_id, $requester, $partof, $status, $remark;
    public bool $showErrorBag = true;

    public function setUp(): array
    {
        $this->showCheckBox();
        $this->persist(['columns', 'filters']);

        return [
            Exportable::make('export')
                ->striped()
                ->type(Exportable::TYPE_XLS, Exportable::TYPE_CSV),

            Header::make()
                ->showSearchInput()
                ->showToggleColumns(),

            Footer::make()
                ->showPerPage()
                ->showRecordCount(),
        ];
    }

    public function datasource(): Builder
    {
        return PurchaseRequest::leftJoin('projects', 'purchase_requests.project_id', '=', 'projects.id')
            ->leftJoin('warehouses', 'purchase_requests.warehouse_id', '=', 'warehouses.id')
            ->select('purchase_requests.*', 'projects.name as project_name', 'warehouses.name as warehouse_name');
    }

    public function relationSearch(): array
    {
        return [];
    }

    public function addColumns(): PowerGridEloquent
    {
        return PowerGrid::eloquent()
            ->addColumn('pr_no')
            ->addColumn('pr_type')
            ->addColumn('project_id')
            ->addColumn('requester')
            ->addColumn('partof')
            ->addColumn('status')
            ->addColumn('remark');
    }

    public function columns(): array
    {
        return [
            Column::make('NO PURCHASE REQUEST', 'pr_no')
                ->sortable()
                ->searchable()
                ->makeInputText()
                ->headerAttribute('', 'width: 15%;'),

            Column::make('TYPE', 'pr_type')
                ->sortable()
                ->searchable()
                ->makeInputText()
                ->headerAttribute('', 'width: 10%;'),

            Column::make('PROJECT', 'project_name')
                ->sortable()
                ->searchable()
                ->makeInputText()
                ->headerAttribute('', 'width: 15%;'),

            Column::make('REQUESTER', 'requester')
                ->sortable()
                ->searchable()
                ->makeInputText()
                ->headerAttribute('', 'width: 10%;'),

            Column::make('PART OF', 'partof')
                ->sortable()
                ->searchable()
                ->makeInputText()
                ->headerAttribute('', 'width: 10%;'),

            Column::make('STATUS', 'status')
                ->sortable()
                ->searchable()
                ->makeInputText()
                ->headerAttribute('', 'width: 5%;'),

            Column::make('NOTE', 'remark')
                ->sortable()
                ->searchable()
                ->makeInputText()
                ->headerAttribute('', 'width: 10%;'),

        ];
    }

    public function actions(): array
    {
        return [

            "class" => "block",

            Button::add('edit')
                ->caption('Edit')
                ->class('text-green-500', 'text-sm', 'width: 25%;')
                ->route('purchase-request-add-item', ['slug' => 'slug'])
                ->target('_self'),

            Button::add('create-po')
                ->caption('Create PO')
                ->class('text-blue-500', 'text-sm')
                ->route('purchase-order-create', ['slug' => 'slug'])
                ->target('_self'),

            Button::add('cancel')
                ->caption('<div>Cancel<div>')
                ->class('text-red-500', 'text-sm')
                ->emit("cancel", ['id' => 'id']),

            Button::add('duplicate')
                ->caption('Duplicate')
                ->class('text-orange-500', 'text-sm')
                ->emit("duplicate", ['id' => 'id']),

            Button::add('info')
                ->caption('Info')
                ->class('text-gray-500', 'text-sm')
                ->emit("open_info", ["id" => "id"]),
        ];
    }

    public function actionRules(): array
    {
        return [
            Rule::button('info')
                ->when(fn () => true)
                ->setAttribute('x-on:click', 'open_info = true'),

            Rule::button('edit')
                ->when(fn ($purchase_request) => count($purchase_request->detail) == 0)
                ->caption('Add Item'),

            Rule::button('edit')
                ->when(fn ($purchase_request) => $purchase_request->status == 'Cancel')
                ->hide(),

            Rule::button('edit')
                ->when(fn ($purchase_request) => $purchase_request->status == 'Duplicated')
                ->hide(),

            Rule::button('edit')
                ->when(fn ($purchase_request) => $purchase_request->status == 'Draft')
                ->hide(),

            Rule::button('cancel')
                ->when(fn ($purchase_request) => $purchase_request->status == 'Cancel')
                ->hide(),

            Rule::button('cancel')
                ->when(fn ($purchase_request) => $purchase_request->status == 'Duplicated')
                ->hide(),

            Rule::button('cancel')
                ->when(fn ($purchase_request) => $purchase_request->status == 'Draft')
                ->hide(),

            Rule::button('duplicate')
                ->when(fn ($purchase_request) => $purchase_request->status != 'Cancel')
                ->hide(),

            Rule::button('create-po')
                ->when(fn ($purchase_request) => count($purchase_request->detail) == 0 || $purchase_request->status == 'Cancel')
                ->hide(),

            Rule::button('create-po')
                ->when(fn ($purchase_request) => $purchase_request->status == 'Duplicated')
                ->hide(),

            Rule::button('create-po')
                ->when(fn ($purchase_request) => $purchase_request->status == 'Draft')
                ->hide(),

            Rule::rows()
                ->when(fn ($purchase_request) => $purchase_request->id == $purchase_request->id)
                ->setAttribute('class', 'flex-col'),

        ];
    }

    public function onUpdatedEditable(string $id, string $field, string $value): void
    {
        $this->validate([
            $field . '.*' => ['required', 'string', 'max:255'],
        ]);

        PurchaseRequest::query()->find($id)->update([
            $field => $value,
        ]);
    }
}
`

**Comment by zaks-zaiya** [comment]
I am also face the same problem.
The Documentation have not explained about this,

Anyone can help?
